spec/controllers/auth/registrations_controller_spec.rb

```diff
@@ -244,9 +244,26 @@
       end
     end
 
-    it 'does nothing if user already exists' do
-      Fabricate(:account, username: 'test')
-      subject
+    context 'with an already taken username' do
+      subject do
+        Setting.registrations_mode = 'open'
+        post :create, params: { user: { account_attributes: { username: 'test' }, email: 'test@example.com', password: '12345678', password_confirmation: '12345678', agreement: 'true' } }
+      end
+
+      before do
+        Fabricate(:account, username: 'test')
+      end
+
+      it 'responds with an error message about the username' do
+        subject
+
+        expect(response).to have_http_status(:success)
+        expect(username_error_text).to match(/has already been taken/)
```

Comment:
It should work, but I'm not comfortable with matching translated strings this way.